### ISSUE TITLE
`renv.download.headers` is ignored during bootstrap

### ISSUE THREAD

**Issue by AlexandreGuinaudeau** [rstudio/renv#1084]
We're using authenticated CRAN channels, so an authentication header must be provided every time a packages is installed.

The `renv.download.headers` works perfectly when using `renv::install`, however it is ignored in [bootstrap.R](https://github.com/rstudio/renv/blob/main/R/bootstrap.R), so bootstrapping fails with:
```
# Bootstrapping renv 0.15.5 --------------------------------------------------
Warning: unable to access index for repository https://<internal-cran-channel>/src/contrib:
  cannot open URL 'https://<internal-cran-channel>/src/contrib/PACKAGES'
Warning: unable to access index for repository https://cloud.r-project.org/src/contrib:
  cannot open URL 'https://cloud.r-project.org/src/contrib/PACKAGES'
* Downloading renv 0.15.5 ... FAILED
```

Why doesn't bootstrap.R use the same logic as download.R to install renv? Is there a way to configure the behavior of the bootstrapping phase without asking users to manually modify the activate.R file that's generated upon activation?

**Comment by kevinushey** [comment]
> Why doesn't bootstrap.R use the same logic as download.R to install renv?

Just because the bootstrap script needs to run in a context where `renv` itself is not available, and this part wasn't included in the bootstrap scripts. I'll see if we can add it in.

**Comment by kevinushey** [comment]
Should now be fixed in the development version of `renv` -- can you please test and let me know if things are working on your side?

**Comment by AlexandreGuinaudeau** [comment]
Thanks for such a quick fix, confirming it works now!

**Comment by kevinushey** [comment]
Great, thanks for confirming!

**Comment by siri16100** [comment]
hi @kevinushey  I use `renv` together with artifactory and also had problems when bootstrapping renv. I debugged into `renv\activate.R`  and found the problem in function `renv_bootstrap_download_cran_latest_find`, specifically at the part where `renv` tries to retrieve the package database: 
```
utils::available.packages(type = type, repos = repos),
```
this is in my case not possible as no headers are provided in function `available.packages`. I changed the line to: 
```
utils::available.packages(type = type, repos = repos, headers = renv_bootstrap_download_custom_headers(repos))
```
and all worked fine. Maybe you have the chance to look into this?
